web/src/app/(main)/games/page.tsx

```diff
@@ -70,60 +98,117 @@ export default function Games() {
-                        <Masonry
-                            columns={{
-                                xs: 1,
-                                sm: 2,
-                                md: 4,
-                                lg: 5,
-                                xl: 6,
-                            }}
-                            spacing={2}
-                        >
-                            {games[key].map((game, index) => (
-                                <GameCard
-                                    key={game.slug}
-                                    game={game}
-                                    index={index}
-                                    localFavorites={localFavorites}
-                                    setLocalFavorites={setLocalFavorites}
-                                />
-                            ))}
-                        </Masonry>
-                    </Box>
-                ))}
+                .map((key) => {
+                    const filteredGames = games[key].filter((game) =>
+                        searchString
+                            ? game.name
+                                  .toLowerCase()
+                                  .includes(searchString.toLowerCase()) ||
+                              game.slug
+                                  .toLowerCase()
+                                  .includes(searchString.toLowerCase())
+                            : true,
+                    );
+
+                    if (filteredGames.length === 0) return null;
+
+                    // this is always defined because we initialize the value, alltho even if it wasn't we'd still get a falsy value
+                    const isCollapsed = collapsedSections![key];
+                    return (
+                        <Box key={key} sx={{ mb: 4 }}>
+                            <Box
+                                sx={{
+                                    display: 'flex',
+                                    alignItems: 'center',
+                                    cursor: 'pointer',
+                                    mb: 1,
+                                    gap: 1,
+                                }}
+                                onClick={() => toggleSection(key as GamesKeys)}
+                            >
+                                <IconButton size="small" sx={{ p: 0.5 }}>
+                                    {isCollapsed ? <ExpandMore /> : <ExpandLess />}
+                                </IconButton>
+                                <Typography variant="h5" sx={{ lineHeight: 1 }}>
+                                    {key}
+                                </Typography>
+                            </Box>
+                            <Collapse
+                                in={!isCollapsed}
+                                timeout="auto"
+                                unmountOnExit
+                            >
+                                <Box
```

Comment:
Why are you using a display: grid box to wrapping Grid components, rather than a Grid component wrapping Item components? The Grid component does use flexbox behind the scenes, but I don't think there's any reason to use need actual CSS grid here